lemmas about powers of elements

jlpaca opened the pull request:
was working on some other stuff that might not be ready for a PR yet, but found myself in need of these & thought they'd be useful.

TO CONTRIBUTORS:

Make sure you have:

  * [ ] reviewed and applied the coding style: [coding](https://github.com/leanprover/mathlib/blob/master/docs/contribute/style.md), [naming](https://github.com/leanprover/mathlib/blob/master/docs/contribute/naming.md)
  * [x] reviewed and applied [the documentation requirements](https://github.com/leanprover/mathlib/blob/master/docs/contribute/doc.md)
  * [ ] for tactics:
     * [ ] added or adapted documentation in [tactics.md](https://github.com/leanprover/mathlib/blob/master/docs/tactics.md)
     * [ ] write an example of use of the new feature in [tactics.lean](https://github.com/leanprover/mathlib/blob/master/test/tactics.lean)
  * [ ] make sure definitions and lemmas are put in the right files
  * [ ] make sure definitions and lemmas are not redundant

If this PR is related to a discussion on Zulip, please include a link in the discussion.

For reviewers: [code review check list](https://github.com/leanprover/mathlib/blob/master/docs/contribute/code-review.md)

jcommelin commented on src/algebra/archimedean.lean:
```diff
@@ -82,6 +86,17 @@ let ⟨M, hM⟩ := pow_unbounded_of_one_lt x hy in
 let ⟨n, hn₁, hn₂⟩ := int.exists_greatest_of_bdd hb he in
   ⟨n, hn₁, lt_of_not_ge (λ hge, not_le_of_gt (int.lt_succ _) (hn₂ _ hge))⟩
 
+/-- The same as exists_int_pow_near, but with ≤ and < the other way around. -/
```
This docstring shows up whenever someone hovers over `exists_int_pow_near'` with their mouse in VScode (in an arbitrary file). But then your sentence seems a bit out of context. It would hence be better if you could write specialised docstrings for each declaration that make sense outside the context of this file.

jlpaca replied:
edited docstrings. each lemmas also now has a comment that points at the other.

jlpaca committed:
feat(algebra/archimedean): add alternative version of exists_int_pow_near

- also add docstrings

jlpaca committed:
feat(analysis/normed_space/basic): additional inequality lemmas

- that there exists elements with large and small norms in a nondiscrete normed field.

jlpaca committed:
doc(algebra/archimedean): edit docstrings

ChrisHughes24 commented on src/algebra/archimedean.lean:
```diff
@@ -67,6 +69,9 @@ end linear_ordered_ring
 section linear_ordered_field
 
+/-- Every positive x is between two successive integer powers of
+another y greater than one. This is the same as exists_int_pow_near',
```
```suggestion
another y greater than one. This is the same as `exists_int_pow_near'`,
```

ChrisHughes24 commented on src/algebra/archimedean.lean:
```diff
@@ -82,6 +87,19 @@ let ⟨M, hM⟩ := pow_unbounded_of_one_lt x hy in
   ⟨n, hn₁, lt_of_not_ge (λ hge, not_le_of_gt (int.lt_succ _) (hn₂ _ hge))⟩
 
+/-- Every positive x is between two successive integer powers of
+another y greater than one. This is the same as exists_int_pow_near,
```
```suggestion
another y greater than one. This is the same as `exists_int_pow_near`,
```

jcommelin committed:
Apply suggestions from code review

Co-Authored-By: Chris Hughes <33847686+ChrisHughes24@users.noreply.github.com>

mergify[bot] committed:
Merge branch 'master' into pow-lemmas
